src/main/java/tahafurkan/sandbox/usermanagement/service/UserService.java

```diff
@@ -12,38 +12,41 @@
 
 @Service
 public class UserService {
-
     @Autowired
     private UserRepository userRepository;
 
     public User get(int id) {
         return userRepository.findById(id).orElseThrow(() -> new NoSuchUserExistsException("NO USER PRESENT WITH ID = " + id));
     }
-    public List<User>  getAll(){
+
+    public List<User> getAll() {
         return userRepository.findAll();
     }
+    public User read(String userName){
+        return userRepository.findByUserName(userName);
+    }
 
     public User create(User user) {
-        User existingUser = userRepository.findById(user.getId()).orElse(null);
-        if (existingUser != null) {
+        boolean userAlreadyExist = userRepository.existsByUserName(user.getUserName());
+        if (userAlreadyExist) {
             throw new UserAlreadyExistsException("User already exists");
         }
+
         return userRepository.save(user);
     }
 
     public User update(int id, User user) {
-        User existingUser = userRepository.findById(id).orElse(null);
-        if (existingUser == null) {
-            throw new NoSuchUserExistsException("No Such User exists :" + id);
-        }
```

Comment:
Bu degisiklikleri zaten gecenki pull request ile gondermemis miydin, neden burada gorunuyor?

Reply:
Eger bu sorunun cevabini bilmedigin icin PR'nin altindaki `This branch has conflicts that must be resolved` mesajiyla karsilasacaksin merge etmeye kalktiginda. Bu bazi durumlarda basit olabilecegi gibi, gercek hayatta basini fazlasiyla agritacaktir.

Reply:
Yazdigim yorumlarin altina cevabini bilmiyorsan, bilmiyorum diye belirt.

Reply:
anladim tamam

Reply:
issue-24 icin yaptigim islemler dogrumu merge ediyim mi?

Reply:
This branch has conflicts that must be resolved bu mesaji nasil giderecegim peki?

Reply:
Ben conflict'i resolve ettim. Bundan sonra, bir issue'ye baslayacagin zaman, lokalinde master branch'deyken `git pull origin master` ile github'daki(remote'daki yani) degisiklikleri al.
Cunku merge burada yapiliyor, lokalinin(yani lokalindeki master'in) bundan haberi yok.

Bu yazdiklarimi da bir yere not et ve unutma, her zaman karsina cikacak bir durum bu.

Reply:
Tamamdir.